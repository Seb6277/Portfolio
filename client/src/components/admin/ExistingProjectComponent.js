import React from 'react'
import {Button} from 'reactstrap'

class ExistingProjectComponent extends React.Component{
  render() {
    return(
      <div className="suppress_project">
        <Button
          className="suppress_button"
          color="danger"
          onClick={() => {if (window.confirm('Are you sure ?')) this.props.deleteAction(this.props.id)}}
        >
          X
        </Button>
        <div className="project_name">
          {this.props.name}
        </div>
        <hr/>
      </div>
    )
  }
}

export default ExistingProjectComponent