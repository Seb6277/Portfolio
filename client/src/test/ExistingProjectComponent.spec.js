import ExistingProjectComponent from '../components/admin/ExistingProjectComponent'

describe('<ExistingProjectComponent/>', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ExistingProjectComponent/>)
    expect(wrapper).toMatchSnapshot()
  });

  it('should call handleDeleteClick on click', () => {
    const spy = sinon.spy();
    const wrapper = mount(<ExistingProjectComponent key={10} name="test" id={10} deleteAction={spy}/>)
    expect(wrapper).toMatchSnapshot()

    wrapper.find("button").first().simulate("click")

    expect(spy.calledOnce).toBe(true)
  });
})