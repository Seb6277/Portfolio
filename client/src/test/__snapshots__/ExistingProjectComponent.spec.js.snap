// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExistingProjectComponent/> should call handleDeleteClick on click 1`] = `
<div
  className="suppress_project"
>
  <button
    aria-label={null}
    className="suppress_button btn btn-danger"
    onClick={[Function]}
    type="button"
  >
    X
  </button>
  <div
    className="project_name"
  >
    test
  </div>
  <hr />
</div>
`;

exports[`<ExistingProjectComponent/> should render correctly 1`] = `
<div
  className="suppress_project"
>
  <Button
    className="suppress_button"
    color="danger"
    onClick={[Function]}
    tag="button"
  >
    X
  </Button>
  <div
    className="project_name"
  />
  <hr />
</div>
`;
